server/src/model.ts

```diff
@@ -6093,3 +6409,572 @@ export async function getPreferredFolderView(loggedInUserId: Uint8Array) {
     select: { cardView: true },
   });
 }
+
+export async function getLibraryStatus({
+  id,
+  userId,
+}: {
+  id: Uint8Array;
+  userId: Uint8Array;
+}): Promise<LibraryInfo> {
+  const info = await prisma.libraryActivityInfos.findUnique({
+    where: {
+      sourceId: id,
+    },
+    select: {
+      status: true,
+      comments: true,
+      sourceId: true,
+      activityId: true,
+    },
+  });
+
+  if (!info) {
+    return { status: "none", sourceId: id, activityId: null };
+  } else {
+    const isAdmin = await prisma.users.findFirst({
+      where: {
+        userId,
+        isAdmin: true,
+      },
+      select: {
+        userId: true,
+      },
+    });
+
+    if (isAdmin) {
+      if (info.activityId) {
+        return info;
+      } else {
+        return {
+          status: info.status,
+          comments: info.comments,
+          sourceId: info.sourceId,
+          activityId: null,
+        };
+      }
+    } else {
+      const isOwner = await prisma.users.findUnique({
+        where: {
+          userId,
+          content: {
+            some: {},
+          },
+        },
```

Comment:
Might as well include a `{select: {userId: true}}` so that it doesn't select all fields.